addons/spreadsheet_account/models/account.py

```diff
@@ -58,16 +58,22 @@ def _build_spreadsheet_formula_domain(self, formula_params):
             ("date", ">=", start),
             ("date", "<=", end),
         ]
+        # It is more optimized to (like) search for code directly in account.account than in account_move_line
         code_domain = expression.OR(
             [
-                expression.AND([
-                    [("account_id.code", "=like", f"{code}%")],
-                    expression.OR([balance_domain, pnl_domain]),
-                ])
-                for code in codes
+                ("code", "=like", f"{code}%"),
             ]
+            for code in codes
         )
-        domain = expression.AND([code_domain, [("company_id", "=", company_id)]])
+        full_codes = self.env["account.account"].search(code_domain).read(["code"])
+        code_domain = expression.OR(
+            [
+                ("account_id.code", "=", f"{code_dict['code']}"),
+            ]
+            for code_dict in full_codes
+        )
+        initial_balance_domain = expression.OR([balance_domain, pnl_domain])
```

Comment:
I'm not really sure what the means "initial balance" here. What do you think of `time_domain` or `period_domain` ?

Reply:
period_domain looks more understandable